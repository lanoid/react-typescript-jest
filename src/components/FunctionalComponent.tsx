import * as React from "react";
import SimpleContext from "../contexts/SimpleContext";

interface Props {
    name?: string,
}

const FunctionalComponent: React.FunctionComponent<Props> = (props) => {
    return <SimpleContext.Consumer>{({ userId }) => { return <p id={userId}>How are you {props.name}?</p>}}</SimpleContext.Consumer>
}

FunctionalComponent.defaultProps = {
    name: "lanoid"
}

export default FunctionalComponent;