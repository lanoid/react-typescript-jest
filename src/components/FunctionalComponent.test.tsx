import * as React from 'react';
import { shallow } from 'enzyme';
import FunctionalComponent from './FunctionalComponent';

describe('FunctionalComponent', () => {
    const functionalComponent = shallow(<FunctionalComponent />);
    
    it('matches snapshot', () => {
        expect(functionalComponent).toMatchSnapshot();
    });
});