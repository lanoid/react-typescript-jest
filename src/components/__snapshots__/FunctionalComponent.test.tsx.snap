// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FunctionalComponent /> matches snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
